ledger/participant-integration-api/src/main/resources/db/migration/oracle/V20__etq_data_migration.sql

```diff
@@ -0,0 +1,125 @@
+------------------------------------ ETQ Data migration -------------------------------
+
+-- Removes all elements from a that are present in b, essentially computes a - b.
+CREATE OR REPLACE FUNCTION etq_array_diff(
+    arrayClob1 IN CLOB,
+    arrayClob2 IN CLOB
+)
+RETURN CLOB
+IS
+        arrayJson1 json_array_t := json_array_t.parse(arrayClob1);
+        outputJsonArray json_array_t := json_array_t ('[]');
+        -- Number type has
+        --  999...(38 9's) x10^125 maximum value
+        -- -999...(38 9's) x10^125 minimum value
+        -- so 200 characters should be enough to hold it together with the whole filter expression
+        filterExpression varchar2(200);
+BEGIN
+    FOR i IN 0 .. arrayJson1.get_size - 1
+    LOOP
+        -- `$[*]` selects each element of the array
+        -- `(@ == v)` is a filter expression that check whether each matched element is equal to some value `v`
+        filterExpression := '$[*]?(@ == ' || (arrayJson1.get(i).to_clob()) ||')';
+        IF NOT json_exists(arrayClob2, filterExpression)
```

Comment:
I fear that this json parsing function in an inner loop may spell trouble. We really need to test this on a sizeable db